src/rabbit_channel.erl

```diff
@@ -789,9 +792,17 @@ check_internal_exchange(#exchange{name = Name, internal = true}) ->
 check_internal_exchange(_) ->
     ok.
 
-check_topic_authorisation(#exchange{name = Name, type = topic}, #ch{user = User}, RoutingKey, Permission) ->
+check_topic_authorisation(#exchange{name = Name = #resource{virtual_host = VHost}, type = topic},
+                          #ch{user = User = #user{username = Username}, conn_pid = ConnPid},
+                          RoutingKey,
+                          Permission) ->
     Resource = Name#resource{kind = topic},
-    Context = #{routing_key => RoutingKey},
+    VariableMap = build_topic_variable_map(
+        amqp_connection:info(ConnPid, [amqp_params]),
```

Comment:
Let's also not use the timeout of `infinity` (I will file a separate issue for `amqp_gen_connection`).